web/src/main/java/org/fao/geonet/monitor/link/LinkCheckerDefault.java

```diff
@@ -23,4 +32,8 @@ public boolean check() {
     public String toString() {
         return url;
     }
+
+    public void setLinkTypes(Set linkTypes) {
```

Comment:
Sorry, been a while since I've programmed in true java - but couldn't the method signature be changed to `public void setLinkTypes(Set<String> linkTypes)`, forcing the code to be correct at compile time, and avoid the possibility of a `ClassCastException` below at runtime?